### PR TITLE
[To feature] Upgrading the version of python on met api

### PR DESCRIPTION
**Pull request by VineetBala-AOT**
Issue #: https://apps.itsm.gov.bc.ca/jira/browse/DESENG-466

*Description of changes:*
- Upgrading the version of python to 3.12
- Upgrading the dependencies to support the version of python.
- Modify code as needed if the newer version does not support the former or lint errors.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-api/src/met_api/services/cdogs_api_service.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ def generate_document(self, template_hash_code: str, data, options):
 
     @staticmethod
     def _post_generate_document(json_request_body, headers, url):
+        # pylint: disable=missing-timeout
         response = requests.post(url, data=json_request_body, headers=headers)
```
Comment (Baelx) [inline met-api/src/met_api/services/cdogs_api_service.py:58]: Was there a reason we couldn't remedy this linter issue? Even if we don't want to specify a timeout value, couldn't we just set it to None?

```suggestion
        response = requests.post(url, data=json_request_body, headers=headers, timeout=None)
```

File: met-api/src/met_api/services/poll_response_service.py [hunk 2]
```diff
@@ -83,6 +84,7 @@ def get_poll_details_with_response_counts(poll_id):
             db.session.query(
                 PollAnswer.id.label('answer_id'),
                 PollAnswer.answer_text,
+                # pylint: disable=not-callable
                 func.count(PollResponseModel.selected_answer_id).label('response_count')
```
Comment (Baelx) [inline met-api/src/met_api/services/poll_response_service.py:88]: From what I've read, it's a known issue for pylint to be labelling `func` method calls as false positives for this rule. You may be able to get around the issue with this stackoverflow answer: https://stackoverflow.com/a/75789407

**Review by Baelx — APPROVED** [review]
Approved! You may just want to review some ways to correct those linter errors instead of putting in exceptions, if possible.

File: met-api/src/met_api/services/cdogs_api_service.py (hunk 1 above)
Comment (VineetBala-AOT) [reply met-api/src/met_api/services/cdogs_api_service.py]: Updated as suggested, thanks.

**Pull request merged**